wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubApiClient.java

```diff
@@ -132,6 +147,47 @@ public GithubPullRequest getPullRequest(
         });
   }
 
+  /**
+   * Returns the latest commit of the branch.
+   *
+   * <p>GitHub REST API documentation: https://docs.github.com/en/rest/commits/commits
+   *
+   * @param user user or organization name
+   * @param repository repository name
+   * @param branch required branch
+   * @param authenticationToken OAuth access token, can be NULL as the GitHub can handle some
+   *     requests without authentication
+   * @return the latest commit of the branch
+   */
+  public GithubCommit getLatestCommit(
+      String user, String repository, String branch, @Nullable String authenticationToken)
```

Comment:
why `authenticationToken` is `@Nullable`?

Reply:
I think we have already discussed it here https://github.com/eclipse-che/che-server/pull/376#discussion_r1006657989

Reply:
https://github.com/eclipse-che/che-server/pull/376#discussion_r1006657989 is about the common `buildGithubApiRequest()` which is used by all the API methods, but I wonder if this particular `getLatestCommit` request can work without authentication?

Reply:
yes, it can

Reply:
If so then why do we need the `authenticationToken` at all?

Reply:
I think we should keep it for private repositories